tests/functional/tests/smoke.py

```diff
@@ -17,7 +17,9 @@ def setUpClass(cls) -> None:
         zaza.get_or_create_libjuju_thread()
         cls.create_local_share_folder()
         cls.model_name = zaza.model.get_juju_model()
+        cls.package_installed = False
         cls.install_package()
+        cls.package_installed = True
```

Comment:
Since this variable is only related to snap, maybe we should rename it to `cls.snap_installed` and move setting it to `True` to be under the snap installation part of `install_package()` method? This way, we can also simplify the condition in `remove_snap_package()` method to be `if cls.snap_installed:`